src/modules/accounting/accounting.py

```diff
@@ -0,0 +1,222 @@
+
+        if not variables.ACCOUNT:
+            logger.info({'msg': 'No account provided to submit extra data. Dry mode'})
+            return
+        if self.w3.transaction.check_transaction(tx, variables.ACCOUNT.address):
+            self.w3.transaction.sign_and_send_transaction(tx, variables.GAS_LIMIT, variables.ACCOUNT)
+
+    # Consensus module: main build report method
+    @lru_cache(maxsize=1)
+    def build_report(self, blockstamp: BlockStamp) -> tuple:
+        report_data = self._calculate_report(blockstamp)
+        logger.info({'msg': 'Calculate report for accounting module.', 'value': report_data})
+        return report_data.as_tuple()
+
+    # # Consensus module: if contract got report data
+    def is_main_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        logger.info({'msg': 'Check if main data was submitted.', 'value': processing_state.main_data_submitted})
+        return processing_state.main_data_submitted
+
+    # Consensus module: if contract could accept any sort of report
+    def is_contract_reportable(self, blockstamp: BlockStamp) -> bool:
+        is_reportable = not self.is_main_data_submitted(blockstamp) or not self.is_extra_data_submitted(blockstamp)
+        logger.info({'msg': 'Check if contract could accept report.', 'value': is_reportable})
+        return is_reportable
+
+    @lru_cache(maxsize=1)
+    def _get_processing_state(self, blockstamp: BlockStamp) -> ProcessingState:
+        ps = named_tuple_to_dataclass(
+            self.report_contract.functions.getProcessingState().call(block_identifier=blockstamp.block_hash),
+            ProcessingState,
+        )
+        logger.info({'msg': 'Fetch processing state.', 'value': ps})
+        return ps
+
+    def is_extra_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        return processing_state.extra_data_items_count == processing_state.extra_data_items_submitted
+
+    def _calculate_report(self, blockstamp: BlockStamp) -> ReportData:
+        validators_count, cl_balance = self._get_consensus_lido_state(blockstamp)
+
+        finalization_share_rate = self._get_finalization_shares_rate(blockstamp)
+        last_withdrawal_id_to_finalize = self._get_last_withdrawal_request_to_finalize(blockstamp)
+
+        exited_validators = self.lido_validator_state_service.get_lido_newly_exited_validators(blockstamp)
+
+        # Here report all exited validators even they were reported before.
+        if exited_validators:
+            node_operator_global_ids_list, exit_validators_count_list = zip(*exited_validators.items())
+            staking_module_ids_list = map(lambda no_global_id: no_global_id[0], node_operator_global_ids_list)
+        else:
+            staking_module_ids_list = exit_validators_count_list = []
```

Comment:
1) we can't relay on `exited_validators` here since `get_lido_newly_exited_validators` returns only changed values, but we need to report total exited validators by module
2) `exit_validators_count_list` should be a map with total exited keys by modules

Let's look at an example:

Current contract state:
Module 1: total exited keys: 10
Module 2: total exited keys: 15
Module 3: total exited keys: 20

CL state: 
Module 1: total exited keys: 12
Module 2: total exited keys: 15
Module 3: total exited keys: 21

2 newly exited validators appears in `Module 1` (10->12), and 1 in `Module 3` (20 -> 21)

This means we have to report:
`stacking_module_id_with_exited_validators=[1,3]`
`count_exited_validators_by_stacking_module=[12,21]`

We do not report Module 2 since there was no change